src/http/server.c

```diff
@@ -241,6 +241,46 @@ static void connect_handler(const struct sa *peer, void *arg)
 }
 
 
+/**
+ * Create an HTTP socket from file descriptor
+ *
+ * @param sockp Pointer to returned HTTP Socket
+ * @param fd    File descriptor
+ * @param reqh  Request handler
+ * @param arg   Handler argument
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+int http_listen_fd(struct http_sock **sockp, re_sock_t fd, http_req_h *reqh,
+		   void *arg)
```

Comment:
if you want to make the diff smaller, you can submit this function as a separate PR ...